### PR TITLE
points calculation and views to simplify querying

### PR DESCRIPTION
**Pull request by nokacper24**

### CODE REVIEW

File: db/migrations/000001_init.up.sql [hunk 1]
```diff
@@ -181,6 +181,34 @@ CREATE TABLE IF NOT EXISTS user_answers (
         REFERENCES answer_alternatives(question_id, id)
 );
 
+-- calculates points awarded for a question based on the time spent on the question, question max points and duration
```
Comment (z-rayc) [inline db/migrations/000001_init.up.sql:184]: Comment says "time spent on the question" and "duration"; what are the differences?

**Review by z-rayc — APPROVED** [review]

File: db/migrations/000001_init.up.sql (hunk 1 above)
Comment (nokacper24) [reply db/migrations/000001_init.up.sql:184]: by duration i mean the question duration, refers to `time_limit_seconds` on the questions table. can clarify

File: db/migrations/000001_init.up.sql [hunk 2]
```diff
@@ -212,4 +240,53 @@ CREATE TABLE IF NOT EXISTS preassigned_roles(
...
+uqp.quiz_id,
+SUM(uqp.points_awarded) AS total_points_awarded,
+ic.is_completed,
+MAX(uqp.answered_at) AS finished_at,
+MAX(uqp.answered_at) < qz.active_to AS answered_within_active_time
+FROM user_question_points uqp
+JOIN questions q ON uqp.question_id = q.id
+JOIN quizzes qz ON q.quiz_id = qz.id,
+LATERAL (
+    -- check if there are qeustions user has not answered in the quiz
```
Comment (Molnes) [inline db/migrations/000001_init.up.sql:277]: use autocorrect, misspelled questions

**Review by Molnes — CHANGES_REQUESTED** [review]

File: db/migrations/000001_init.up.sql (hunk 2 above)
Comment (nokacper24) [reply db/migrations/000001_init.up.sql]: smh nothing gets through

**Review by Molnes — APPROVED** [review]

**Pull request merged**